src/masternodeman.cpp

```diff
@@ -88,25 +88,26 @@ void CMasternodeMan::AskForMN(CNode* pnode, const COutPoint& outpoint, CConnman&
     CNetMsgMaker msgMaker(pnode->GetSendVersion());
     LOCK(cs);
 
-    std::map<COutPoint, std::map<CNetAddr, int64_t> >::iterator it1 = mWeAskedForMasternodeListEntry.find(outpoint);
+    CService addrSquashed = IsArgSet("-regtest") ? (CService)pnode->addr : CService(pnode->addr, 0);
```

Comment:
What is the reason for using `IsArgSet()` here instead of checking `Params().NetworkIDString()` as we do in other places?
Because of the length of the check or any other reason? I wonder if it's maybe time to have a more generic boolean in chainparams that says something like "IsRegtest", "IsLocalNetwork" or "IsRegtestNetwork" for all these regtest specific network changes.

Whatever we decide, it's not scope of this PR, so still utACK.

Reply:
Yep, `IsArgSet` is just shorter and makes it easier to read imo. Also, we should avoid using this kind of comparison, so I pushed an updated version. Thanks for the reminder :)

Note: original commit was rebased without changes, all new changes are in e9e97c3